### ISSUE TITLE
Add a switch to disable the style checks

### ISSUE THREAD

**Issue by XhmikosR** [validator/validator#585]
I totally get it, since you are just following the specs, and I'm with you on that.

But in our current world, it's impossible to have "valid" CSS, especially for huge apps, critical CSS, etc.

As you already know, browser vendors just do their thing, implementing stuff when they are just announced, or are just CR, adding their own properties and so on.

Here are a few examples:

```
Error: CSS: Unrecognized at-rule “@-ms-viewport”
Error: CSS: “touch-action”: Property “touch-action” doesn't exist:
Error: CSS: “clip-path”: “inset(50%)” is not a “clip-path” value:
```

So, if you could add an option to disable the style checks, that would save a lot of time for many people using the validator that will just end up with many errors, while those errors are actually needed.

**Comment by Arttse** [comment]
Yeah. In general, this is not the whole list of examples, I have met many difficulties. CSS Validator needs a lot of updates. Well, while this happens I think it would be nice to make an option.

**Comment by sideshowbarker** [comment]
I will look into this but I would also appreciate people taking the time to raise issues about any cases where the CSS validator needs updates.

Based on this issue, we already updated it to support vendor-prefixed at-rules like `@-ms-viewport` and to support `touch-action`

**Comment by JamesAndersonJr** [comment]
I'm not seeing the effects of the update yet. Check: https://www.jamesandersonjr.com/pay_james.php

**Comment by sideshowbarker** [comment]
So when I check https://www.jamesandersonjr.com/pay_james.php I get this:

```
CSS: Parse Error in @-ms-viewport↩	{↩		width: auto !important;↩	}
```
The cause of that is a known issue: If a vendor-prefixed at-rule is the first rule in the stylesheet, the CSS checker backend doesn’t parse it correctly. A workaround for now is just to add some empty/inconsequential rule before the vendor-prefixed at-rule:

```html
<style media="all">
#foo { }
@-ms-viewport
	{
		width: auto !important;
	}
</style>
```

**Comment by sideshowbarker** [comment]
@JamesAndersonJr:
>> I'm not seeing the effects of the update yet. Check: https://www.jamesandersonjr.com/pay_james.php
>
> The cause of that is a known issue: If a vendor-prefixed at-rule is the first rule in the stylesheet, the CSS checker backend doesn’t parse it correctly.

I’ve now fixed that.

https://validator.w3.org/nu/?doc=https://www.jamesandersonjr.com/pay_james.php now has no errors.

Thanks for having taken time to report this

**Comment by JamesAndersonJr** [comment]
Thanks!

**Comment by sideshowbarker** [comment]
So I’m not inclined at this point to add the switch requested in this issue. The reason is that after making literally dozens of changes and refinements to the CSS-checking backend of the last couple of months, we now have the underlying CSS-checking behavior to the point that I think the `style`-checking behavior is conformant enough that it’s serving the intended purpose without emitting an excessive amount of false positives for cases that the CSS-checking backend doesn’t support yet.

So at this point my request is that if you have cases where the checker is still emitting spurious false-positive errors when checking a particular document or markup snippet, then please add a comment here with the URL for the document or with a markup snippet, and I’ll make time to investigate and fix things so that the checker no longer emits any spurious false-positive errors for that document/case.

Otherwise, lacking any new data/evidence that the `style`-checking behavior isn’t ready to ship yet, I plan to a new major release vnu.jar very soon, with the `style`-checking behavior included. And by “soon”, I mean this week or next.

**Comment by XhmikosR** [comment]
Sounds good to me. I haven't had the time to check your progress, but your word is enough to me :)

**Comment by Arttse** [comment]
@sideshowbarker I've found a bug.

This code passes:
```css
.example {
  background-position-x123: center;
}
```
But property `background-position-x123` doesn't exist. 
Or something I don't know? :smile:

All passes with this pattern: `background-position-x*`
The same with `background-position-y*`

**Comment by Arttse** [comment]
@sideshowbarker how about this?
```css
@charset "utf-8";

.example {
  opacity: 0;
}
```
```
Error: The @charset rule may only occur at the start of the style sheet. Please check that there are no spaces before it..
```

**Comment by sideshowbarker** [comment]
> This code passes:
```css
.example {
  background-position-x123: center;
}
```
> But property `background-position-x123` doesn't exist. 
> Create a new issue for this?

No need — I’ve already fixed it. Thanks for catching it

**Comment by sideshowbarker** [comment]
```css
@charset "utf-8";

.example {
  opacity: 0;
}
```
>Error: The @charset rule may only occur at the start of the style sheet. Please check that there are no spaces before it..

Fixed now — thanks for catching that

**Comment by cagatay117** [comment]
@sideshowbarker , did you add a switch to disable the style checks? I am fed up with "property doesn't exist" errors :/
